docs/design_docs.md

```diff
@@ -37,34 +36,33 @@ The ERC-20 bridge facilitates the transfer of ERC-20 tokens from Ethereum to be
+1. A transaction is built and submitted by either the user or a relayer service that meets the requirements of the `MessageToFungibleTokenPredicate`
+1. A single call is made from the transaction script to the intended recipient Fuel token contract. This function verifies the sender and predicate owner of the `InputMessage`, parses the data from the `InputMessage` data field and mints the appropriate amount of tokens to the designated recipient
 
 ![ERC20 Deposit Diagram](/docs/imgs/FuelMessagingERC20Deposit.png)
 
 ### ERC-20 Bridge Withdrawal
 
 1. User starts a withdrawal by calling the `FuelMyToken` contract sending some coins to withdraw along with it
-1. `FuelMyToken` contract looks to see what coins it was sent, burns them and then creates a `OutputMessage` via opcode
-1. `OutputMessage` is noted on L1 by including the messagId in a merkle root in the state header committed to L1
-1. After any necessary finalization period, the user calls to the `FuelMessageInbox` with a merkle proof of the previous sent message
-1. The `FuelMessageInbox` verifies the given merkle proof and makes the message call to the `L1ERC20Gatewa`y specified in the message
-1. The `L1ERC20Gateway` verifies it’s being called by the `FuelMessageInbox` and releases the specified amount of tokens to the specified address
+1. The `FuelMyToken` contract looks to see what coins it was sent, burns them and then creates a `OutputMessage` via the `SMO` opcode
+1. All `OutputMessages` are committed to L1 via a merkle tree of all `OutputMessage` messageIds by a designated committer (either a multisig or future consensus contract)
+1. After a finalization period, the user calls to the `FuelMessagePortal` with a merkle proof of the previous sent message which then calls to the `L1ERC20Gateway` with the abi specified in the message data
+1. The `L1ERC20Gateway` verifies it’s being called by the `FuelMessagePortal` and releases the specified amount of tokens to the specified address
 
 ![ERC20 Withdrawal Diagram](/docs/imgs/FuelMessagingERC20Withdraw.png)
 
 ## Retryable Messages
 
 In order to prevent messages getting lost during generic messaging from L1 to Fuel, developers should follow the following standard practice utilizing common libraries.
 
-1. Either a contract or EOA calls `sendMessage` on the `FuelMessageOutbox` that creates a message to be relayed on Fuel with the `MessageToContractPredicate` so that anyone can spend the `InputMessage` on a user's behalf but with guarantees that the transaction is built as it’s supposed to
-1. Client sees the message on L1 via event logs
+1. Either a contract or EOA calls `sendMessage` on the `FuelMessagePortal` that creates a message to be relayed on Fuel with the `MessageToContractPredicate` so that anyone can spend the `InputMessage` on a user's behalf but with guarantees that the transaction is built as it’s supposed to
```

Comment:
```suggestion
1. Either a contract or EOA calls `sendMessage()` on the `FuelMessagePortal` that creates a message to be relayed on Fuel with the `MessageToContractPredicate` so that anyone can spend the `InputMessage` on a user's behalf but with guarantees that the transaction is built as it’s supposed to
```